contracts/ClaimZap.sol

```diff
@@ -0,0 +1,199 @@
+        veAssetDeposit = _veAssetDeposit;
+        ve3TokenRewards = _ve3TokenRewards;
+        ve3dRewards = _ve3dRewards;
+        exchange = _exchange;
+        ve3dLocker = _locker;
+    }
+
+    function getName() external pure returns (string memory) {
+        return "ClaimZap V2.0";
+    }
+
+    function setApprovals() external onlyOwner{
+        IERC20(veAsset).safeApprove(veAssetDeposit, 0);
+        IERC20(veAsset).safeApprove(veAssetDeposit, type(uint256).max);
+        IERC20(veAsset).safeApprove(exchange, 0);
+        IERC20(veAsset).safeApprove(exchange, type(uint256).max);
+
+        IERC20(veToken).safeApprove(ve3dRewards, 0);
+        IERC20(veToken).safeApprove(ve3dRewards, type(uint256).max);
+
+        IERC20(ve3Token).safeApprove(ve3TokenRewards, 0);
+        IERC20(ve3Token).safeApprove(ve3TokenRewards, type(uint256).max);
+
+        IERC20(veToken).safeApprove(ve3dLocker, 0);
+        IERC20(veToken).safeApprove(ve3dLocker, type(uint256).max);
+    }
+
+    function CheckOption(uint256 _mask, uint256 _flag) internal pure returns(bool){
+        return (_mask & (1<<_flag)) != 0;
+    }
+
+    /// @notice Claim rewards from multiple reward pools in one transaction and allow passing options for staking, claiming locked balances etc.
+    /// @param rewardContracts Array of contract addresses to claim rewards from
+    /// @param extraRewardContracts Array of contract addresses to claim extra rewards from
+    /// @param tokenRewardContracts Array of contract addresses to claim multiple reward tokens from
+    /// @param tokenRewardTokens Array of token addresses to claim in multiple token reward contracts
+    /// @param depositVeAssetMaxAmount Maximum amount of VeAsset to lock and stake
+    /// @param minAmountOut Minimum amount to swap in exchange
+    /// @param depositVeTokenMaxAmount Maximum amount of VeToken to stake
+    /// @param options number that represents the options for claiming extras
+    function claimRewards(
+        address[] calldata rewardContracts,
+        address[] calldata extraRewardContracts,
+        address[] calldata tokenRewardContracts,
+        address[] calldata tokenRewardTokens,
+        uint256 depositVeAssetMaxAmount,
+        uint256 minAmountOut,
+        uint256 depositVeTokenMaxAmount,
+        uint256 options
+    ) external{
+        uint256 veAssetBalance = IERC20(veAsset).balanceOf(msg.sender);
+        uint256 veTokenBalance = IERC20(veToken).balanceOf(msg.sender);
+
+        //claim from main LP pools
+        for(uint256 i = 0; i < rewardContracts.length; i++){
+            IBasicRewards(rewardContracts[i]).getReward(msg.sender,true);
+        }
+        //claim from extra rewards
+        for(uint256 i = 0; i < extraRewardContracts.length; i++){
+            IBasicRewards(extraRewardContracts[i]).getReward(msg.sender);
+        }
+        //claim from multi reward token contract
+        for(uint256 i = 0; i < tokenRewardContracts.length; i++){
```

Comment:
what the token reward contracts refer to in our platform?

Reply:
I don't think there is one at the moment, though this contract was meant to be used more in the future. So it's your call whether we should keep this or not

Reply:
understand , but we already have this pools in our platform which partially match this interface , so it makes sense to adjust it to what we already have in VE3DRewardPool and VE3DLocker 
https://github.com/veTokenFinance/veToken-contracts/blob/main/contracts/VE3DRewardPool.sol#L356
https://github.com/veTokenFinance/veToken-contracts/blob/main/contracts/VE3DLocker.sol#L768

Reply:
@c-future so we should claim from VE3DRewardPool and VE3DLocker? Let me know what you want me to do here

Reply:
yes